From 9d55e2740df3ef7a2fe1fa81a553a5c7761e77a0 Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Wed, 23 Oct 2013 13:31:50 -0400
Subject: [PATCH 0562/1089] Prevent kernel panic if USDPAA mem cannot be
 allocated

[Original patch taken from QorIQ-SDK-V1.6-SOURCE-20140619-yocto.iso]

USDPAA DMA memory is allocated very early in the boot process
using the memblock_alloc functions.  The old function would
call panic() in cases that the allocation would fail, but
no indication would be given on the console since the
drivers were not loaded yet.  This patch changes the function
called so that the panic will not occur and debug info can be
seen in the boot log

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
Change-Id: I8ebe9eb826973ea8ef6d16ba6fc73512b779ab77
Reviewed-on: http://git.am.freescale.net:8181/5953
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Haiying Wang <Haiying.Wang@freescale.com>
Reviewed-by: Thorpe Geoff-R01361 <Geoff.Thorpe@freescale.com>
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
---
 drivers/staging/fsl_qbman/fsl_usdpaa.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/fsl_usdpaa.c b/drivers/staging/fsl_qbman/fsl_usdpaa.c
index 7ee4997..3ac3fa0 100644
--- a/drivers/staging/fsl_qbman/fsl_usdpaa.c
+++ b/drivers/staging/fsl_qbman/fsl_usdpaa.c
@@ -1486,7 +1486,9 @@ __init void fsl_usdpaa_init_early(void)
 		phys_size = 0;
 		return;
 	}
-	phys_start = memblock_alloc(phys_size, largest_page_size(phys_size));
+	phys_start = __memblock_alloc_base(phys_size, 
+					   largest_page_size(phys_size),
+					   MEMBLOCK_ALLOC_ACCESSIBLE );
 	if (!phys_start) {
 		pr_err("Failed to reserve USDPAA region (sz:%llx)\n",
 		       phys_size);
-- 
2.0.2

